alf/algorithms/qrsac_algorithm.py

```diff
@@ -0,0 +1,151 @@
+    return alf.nn.Sequential(
+        alf.nn.EncodingNetwork(
+            input_tensor_spec,
+            fc_layer_params=(64, 64),
+            last_layer_size=64,
+            last_activation=alf.math.identity),
+        alf.nn.QuantileProjectionNetwork(
+            input_size=64, output_tensor_spec=TensorSpec((quantile_size, ))))
+
+
+@alf.configurable
+class QrsacAlgorithm(SacAlgorithm):
+    """Quantile regression actor critic algorithm. """
+
+    def __init__(self,
+                 observation_spec,
+                 action_spec: BoundedTensorSpec,
+                 reward_spec=TensorSpec(()),
+                 actor_network_cls=ActorDistributionNetwork,
+                 critic_network_cls=critic_network_cls,
+                 epsilon_greedy=None,
+                 use_entropy_reward=False,
+                 calculate_priority=False,
+                 num_critic_replicas=2,
+                 env=None,
+                 config: TrainerConfig = None,
+                 critic_loss_ctor=None,
+                 target_entropy=None,
+                 prior_actor_ctor=None,
+                 target_kld_per_dim=3.,
+                 initial_log_alpha=0.0,
+                 max_log_alpha=None,
+                 target_update_tau=0.05,
+                 target_update_period=1,
+                 dqda_clipping=None,
+                 actor_optimizer=None,
+                 critic_optimizer=None,
+                 alpha_optimizer=None,
+                 debug_summaries=False,
+                 name="QrsacAlgorithm"):
+        """
+        Refer to SacAlgorithm for Args. Currently only continuous action
+        space is supported.
+        """
+        super().__init__(
+            observation_spec,
+            action_spec,
+            reward_spec=reward_spec,
+            actor_network_cls=actor_network_cls,
+            critic_network_cls=critic_network_cls,
+            epsilon_greedy=epsilon_greedy,
+            use_entropy_reward=use_entropy_reward,
+            calculate_priority=calculate_priority,
+            num_critic_replicas=num_critic_replicas,
+            env=env,
+            config=config,
+            critic_loss_ctor=critic_loss_ctor,
+            target_entropy=target_entropy,
+            prior_actor_ctor=prior_actor_ctor,
+            initial_log_alpha=initial_log_alpha,
+            max_log_alpha=max_log_alpha,
+            target_update_tau=target_update_tau,
+            target_update_period=target_update_period,
+            dqda_clipping=dqda_clipping,
+            actor_optimizer=actor_optimizer,
+            critic_optimizer=critic_optimizer,
+            alpha_optimizer=alpha_optimizer,
+            debug_summaries=debug_summaries,
+            name=name)
+
+        assert self._act_type == ActionType.Continuous, (
+            "Only continuous action space is supported for qrsac algorithm.")
+
+        self._num_quantiles = self._critic_networks.output_spec.shape[-1]
```

Comment:
The `self._num_quantiles` seems to be unused.

Reply:
Removed.